README.md

```diff
@@ -32,22 +33,231 @@ scheduled_task { 'Run Notepad':
+One common use is to kick off a script, either once or on a regular cadence.
+In this first example we schedule a cleanup script to run this one time.
+
+~~~ puppet
+scheduled_task { 'Disk Cleanup': # Unique name for the scheduled task
+  command => 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe -File "C:\\Scripts\Clear-DiskSpace.ps1"',
+  enabled => 'true',             # This is the default, but including it is good practice. Flip to 'false' to disable the task.
+  trigger => [{
+    schedule   => 'once',        # Defines the trigger type; required.
+    start_time => '23:20',       # Defines the time the task should run; required.
+    start_date => '2018-01-01'   # Defaults to the current date; not required.
+  }],
+}
+~~~
+
+If we need to have the cleanup script run every night we can use a daily trigger.
+Just changing the trigger schedule from `once` to `daily` will do the trick.
+
+~~~ puppet
+scheduled_task { 'Disk Cleanup Nightly':
+  command => 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe -File "C:\\Scripts\Clear-DiskSpace.ps1"',
+  enabled => 'true',
+  trigger => [{
+    schedule         => 'daily',
+    start_time       => '23:20'
+  }],
+}
+~~~
+
+You can also set your scheduled tasks to repeat during an set time block.
+Re-using the idea of a cleanup script, this scheduled task begins at the same time every day and runs once an hour from seven in the morning to seven at night as the SYSTEM account.
+
+~~~ puppet
+scheduled_task { 'Disk Cleanup Daily Repeating':
+  ensure  => 'present',
+  command => 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe -File "C:\\Scripts\Clear-DiskSpace.ps1"',
+  enabled => 'true',
+  trigger => [{
+    'schedule'         => 'daily',
+    'start_time'       => '07:00',
+    'minutes_duration' => '720',   # Specifies the length of time, in minutes, the task is active
+    'minutes_interval' => '60'     # Causes the task to run every hour
+  }],
+  user    => 'system',             # Specifies the account to run the task as
+}
+~~~
+
+The downside to that task is that it causes the cleanup script to run _every_ day, even on the weekends when there is no activity.
+We can instead use a weekly trigger to fix this:
+
+~~~puppet
+scheduled_task { 'Disk Cleanup Weekly Repeating':
+  ensure  => 'present',
+  command => 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe -File "C:\\Scripts\Clear-DiskSpace.ps1"',
+  enabled => 'true',
+  trigger => [{
+    'schedule'         => 'weekly',
+    'start_time'       => '07:00',
+    'day_of_week'      => ['mon', 'tues', 'wed', 'thurs', 'fri'], # Note the absence of Sunday and Monday
+    'minutes_interval' => '60',
+    'minutes_duration' => '720'
+  }],
+  user    => 'system',
+}
+~~~
+
+Similarly, we can schedule our cleanup script to run monthly if we decide the cleanup script doesn't need to run as often or is particularly resource intensive.
```

Comment:
Do we need to put any effort into speculating on behalf of the user why they might want to run a task monthly? It's such a common requirement that I think we can probably end this scenario description at "to run monthly" or at "need to run as often".

Reply:
This was more part of a narrative doc writing thing, where you take a fictional character through a chain of events with consistent internal logic.